### PR TITLE
WIP: rename the makeRoot* to add*

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by wbpcode** [comment]
cc @jmarantz here.

File: source/common/json/json_streamer.h [hunk 1]
```diff
@@ -344,25 +344,17 @@ template <class OutputBufferType> class StreamerBase {
...
   /**
-   * Makes a root map for the streamer.
-   *
-   * You must create a root map or array before any of the JSON population
-   * functions can be called, as those are only available on Map and Array
-   * objects.
+   * Adds a root map for the streamer.
    */
-  MapPtr makeRootMap() {
+  MapPtr addMap() {
```
Comment (jmarantz) [inline source/common/json/json_streamer.h:349]: all you are doing is changing the name of this function. It is still restricted be a root level map.

If this works, why can't you just call makeRootMap from the other PR and not do this change?

Comment (wbpcode) [reply source/common/json/json_streamer.h:349]: See here https://github.com/envoyproxy/envoy/pull/36334#discussion_r1777180288

They are basically duplicated. If change the makeRootMap to addMap. I then could deduplicate the code by a simple template. (After this change, the Level and the Streamer provide almost same methods).

**Comment by wbpcode** [comment]
This is unnecessary after a discussion.

**Pull request closed without merging**